### ISSUE TITLE
Accessibility: Cart and checkout page labels fail color contrast

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43619]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Labels for fields on the cart and checkout page are set to hsla(0,0%,7%,.5) which has a 3.52:1 color contrast ratio, which fails WCAG.
<img width="310" alt="Screenshot 2024-01-14 at 7 13 57 PM" src="https://github.com/woocommerce/woocommerce/assets/13153456/f4e7c440-21e9-4ca7-9ec4-17fdbcfe4a91">

### Expected behavior

All field labels should pass color contrast checks at a [minimum of 4.5:1 (WCAG AA)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) but most ideally at a ratio of [7:1 (WCAG AAA)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-enhanced.html), to ensure that people who have low vision can complete checkout.

### Actual behavior

Field labels are light gray and fail contrast.

Here's where the color on the labels is set to $universal-body-low-emphasis:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce-blocks/packages/components/text-input/style.scss#L17C11-L17C38

And where the color is defined:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce-blocks/assets/css/abstracts/_colors.scss#L23

### Steps to reproduce

See code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
